### DISCUSSION TITLE
https://angular.love/why-inline-templates-are-great

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#343] [General]
# https://angular.love/why-inline-templates-are-great

Angular.love - a place for all Angular enthusiasts created to inspire and educate.

https://angular.love/why-inline-templates-are-great

**Comment by bhalperin** [comment]
Thanks for the blog. Bottom line: it's totally subjective. I personally prefer a separate file for the template. I have always favored separating markup from logic, including visual separation in the IDE. Same argument for keeping the stylesheet in a separate file. Still, breaking down components as you nicely demonstrated in the header component sample, is highly recommended for all the reasons you listed.

**Reply by Dyqmin** [reply]
Hi, thanks for the comment! That's correct - it's all about personal preferences. Separate template files are more common and people generally are more familiar with this pattern. I wanted to demystify (or "de-demonize") inline templates because I’ve noticed the efficiency and other benefits they bring since I started using them, and I wanted to share my thoughts. Hope you give them a shot! :)

**Comment by fyodorio** [comment]
I actually like [the AnalogJS’ experimental idea](https://analogjs.org/docs/experimental/sfc) for SFC implementation. I would love to see its wider adoption and further corresponding enhancements.

**Comment by commadelimited** [comment]
This feels a bit like false proof. Sure your main template is now 63% shorter, but that comes at the cost of 4 new components, increase app complexity, and decreased performance. All so that you can inline a template? It's worth pointing out too that a standalone HTML template could more easily be passed off to a UI developer, one who is a wiz with markup, but perhaps isn't as familiar with JavaScript.

**Reply by nathantaal** [reply]
yeah but then again, if you would make 3 small components like button, header and footer, you'll end up with 9 files while you can have 3. And to me, that really decreases app complexity. I really like that fact that I can see how my variables and functions are used within the template, without having to split html/ts next to eachother. Of course, this stops being an advantage when the component becomes an god object. So like mentioned in the article, it comes down to preference. You however mention something about performance, do you have any sources to confirm inline components have worst performance? Because that might be a dealbreaker, but from what I can find there is a slightly (neglectable) performance increase.
